pkg/featuremanager/feature_manager.go

```diff
@@ -0,0 +1,157 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package featuremanager
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"log/slog"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+
+	"github.com/cilium/cilium/pkg/dynamicconfig"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+type FeatureStatus struct {
+	Feature Feature
+	Enabled bool
+}
+
+type FeatureManager interface {
+	Append(feature Feature, hook cell.HookInterface)
+	List() []Feature
+
+	start(feature Feature, ctx context.Context) error
+	stop(feature Feature, ctx context.Context) error
+	process(status []FeatureStatus, ctx context.Context) error
+}
+
+type DefaultFeatureManager struct {
```

Comment:
Are we expecting to have many of these? If not, then maybe it's better to just have `type FeatureManager struct`, and maybe `type FeatureAppender interface` in addition for the cases where we want to be able to override/wrap it.